pkg/netstore/netstore.go

```diff
@@ -23,16 +26,35 @@ type store struct {
 	validator        swarm.Validator
 	logger           logging.Logger
 	recoveryCallback recovery.RecoveryHook // this is the callback to be executed when a chunk fails to be retrieved
+	chunkChanMap     map[string]chan swarm.Chunk
+	chunkChanMu      *sync.RWMutex
 }
 
 var (
-	ErrRecoveryAttempt = errors.New("failed to retrieve chunk, recovery initiated")
+	repairTimeout = 10 * time.Second
+)
+
+var (
+	ErrRecoveryTimeout          = errors.New("timeout during chunk repair")
+	errDecodingRecoveryResponse = errors.New("error decoding recovery response")
 )
 
 // New returns a new NetStore that wraps a given Storer.
 func New(s storage.Storer, rcb recovery.RecoveryHook, r retrieval.Interface, logger logging.Logger,
 	validator swarm.Validator) storage.Storer {
-	return &store{Storer: s, recoveryCallback: rcb, retrieval: r, logger: logger, validator: validator}
+	return &store{
+		Storer:           s,
+		recoveryCallback: rcb,
+		retrieval:        r,
+		logger:           logger,
+		validator:        validator,
+		chunkChanMap:     make(map[string]chan swarm.Chunk),
+		chunkChanMu:      &sync.RWMutex{},
+	}
+}
+
+func SetTimeout(timeout time.Duration) {
```

Comment:
This is called in netstore also, that why it is exported.

> Or maybe, to have repairTimeout as a configurable field in store.

do you mean have it as a configurable value from command line?

Reply:
> This is called in netstore also, that why it is exported.

It is in the `netstore`, that is the point of my comment.

> 
> > Or maybe, to have repairTimeout as a configurable field in store.
> 
> do you mean have it as a configurable value from command line?

No, as I said, the field to the store, so that it can be configured in netstore constructor for tests. That is just another approach instead the package global variable.